src/main/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaMapper.java

```diff
@@ -118,6 +118,9 @@ private void buildRegex() {
                 } else if (part.equals(MQTT_TOPIC_SINGLE_LEVEL_WILDCARD_CHARACTER)) {
                     ruleRegex.append(SINGLE_LEVEL_WILDCARD_REGEX);
                 } else if (part.equals(MQTT_TOPIC_MULTI_LEVEL_WILDCARD_CHARACTER)) {
+                    if (ruleRegex.length() > 1) {
+                        ruleRegex.deleteCharAt(ruleRegex.length() - 1);
```

Comment:
Remove the last `/` before appending `.*` to the regex expression. 
If the pattern is `{house}/#`, the final regex will be `(?<house>[^/]+).*` 
and now `my_house/floor`, `first_house/temperature`, and `my_house` will match.

I have also tried using regex `lookaround`, but was not successful.